apps/marketplace/components/Brief/Overview/BriefOverviewSectionList.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import AUlinklist from '@gov.au/link-list/lib/js/react.js'
+
+import styles from './BriefOverviewSectionList.scss'
+
+export class BriefOverviewSectionListComponent extends Component {
+  static propTypes = {
+    links: PropTypes.array.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      showLinks: false
+    }
+  }
+
+  preventDefault = e => {
+    e.preventDefault()
+  }
+
+  toggle = () => {
+    this.setState({
+      showLinks: !this.state.showLinks
```

Comment:
This one looks pretty benign but in general when setting React state you should not reference the current state using `this.state` - this is because state changes can be asynchronous so two quick fires of this may not be handled properly. The recommendation is to pass `setState` a function, which is given the previous state as an arg, e.g.:

```
this.setState(prevState => ({
  showLinks: !prevState. showLinks
}));
```

Further reading: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
Thank you.  This has been updated so that `setState` is passed a function.